Sources/Algorithms/Unique.swift

```diff
@@ -14,33 +14,57 @@
 //===----------------------------------------------------------------------===//
 
 extension Sequence where Element: Hashable {
-  /// Returns an array with only the unique elements of this sequence, in the
-  /// order of the first occurence of each unique element.
+  /// Returns an array with only  unique elements of this sequence, in the
+  /// order of the first occurrence of each unique element.
+  ///
+  /// In this example, `uniqued()` is used to include only names which are
+  /// unique
+  ///
+  ///     let animals = ["dog", "pig", "cat", "ox", "dog","cat"]
+  ///     let uniqued = animals.uniqued()
+  ///     print(uniqued)
+  ///     // Prints "["dog", "pig","cat", "ox"]"`
+  ///
+  /// - Returns: Returns an array with only the unique elements of this sequence
+  ///  .
+  /// - Complexity: O(*n*), where *n* is the length of the sequence.
   public func uniqued() -> [Element] {
     var seen: Set<Element> = []
     var result: [Element] = []
-    for element in self {
-      if seen.insert(element).inserted {
+    for element in self where seen.insert(element).inserted {
         result.append(element)
-      }
     }
     return result
-  }
+    }
 }
 
 extension Sequence {
   /// Returns an array with the unique elements of this sequence (as determined
-  /// by the given projection), in the order of the first occurence of each
+  /// by the given projection), in the order of the first occurrence of each
   /// unique element.
+  ///
+  /// In this example, `uniqued()` is used to include only names which are
+  /// unique
+  ///
+  ///     let animals = ["dog", "pig", "cat", "ox", "dog","cat"]
+  ///     let uniqued = animals.uniqued()
+  ///     print(uniqued)
+  ///     // Prints "["dog", "pig","cat", "ox"]"`
```

Comment:
Could you switch this to an example that demonstrates the `uniqued(on:)` method? This example shows the parameterless method defined above.